src/lib.rs

```diff
@@ -182,6 +183,55 @@ impl Pixels {
         PixelsBuilder::new(width, height, surface_texture).build()
     }
 
+    /// Resize the pixel buffer, this doesn't resize the surface upon which the pixel buffer is rendered.
+    pub fn resize_buffer(&mut self, width: u32, height: u32) {
+        self.context.texture_extent = wgpu::Extent3d {
+            width,
+            height,
+            depth: 1,
+        };
+
+        // Update ScalingMatrix for mouse transformation
+        self.scaling_matrix_inverse = renderers::ScalingMatrix::new(
+            (
+                self.context.texture_extent.width as f32,
+                self.context.texture_extent.height as f32,
+            ),
+            (width as f32, height as f32),
+        )
+        .transform
+        .inversed();
+
+        // Recreate the texture
```

Comment:
The rest of this function looks a lot like code in the builder. This _could_ be DRY'd, too. That might help with maintainability?

Reply:
Yes, that makes sense. I'll look at it when the part about swap chain creation is stabilized.